### PR TITLE
fix: make field ID compatible with both string and int

### LINKED ISSUE #896: bug: manager api does not support int for field `id`
# Bug report

## Describe the bug

`admin api` supports the id field as string and int type, `manager api` only supports string, 
which will cause the data with int type id entered in `admin api` cannot be displayed with `manager api`

## How to Reproduce

1.  create upstream
```sh
curl "http://127.0.0.1:9080/apisix/admin/upstreams/60" -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -i -X PUT -d '
{
    "type": "roundrobin",
	"name":"cmd-test",
	"desc":"cmd-test-desc",
    "nodes": {
        "127.0.0.1:58080": 1
    }
}'
```

2. use upstream  id as int
```
curl "http://127.0.0.1:9080/apisix/admin/routes/6" -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "uri": "/*",
	"name":"route test",
	"desc":"hello route",
    "host": "127.0.0.1",    
    "upstream_id": 60
}'
```

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

close #896

### CODE REVIEW

**Comment by membphis** [comment]
`api/test/e2e/id_compatible_test.go` looks good for me. I think we can turn this PR to `ready` @nic-chen

**Comment by membphis** [comment]
> `api/test/e2e/id_compatible_test.go` looks good for me

for the source code, need @gxthrj @ShiningRush @tokers @spacewander help ^_^

**Review by gxthrj — APPROVED** [review]
LGTM

**Comment by nic-chen** [comment]
@ShiningRush @tokers @spacewander
 please have a look, thanks.

File: api/internal/core/entity/entity.go [hunk 1]
```diff
@@ -17,24 +17,30 @@
...
 	"time"
 
 	"github.com/apisix/manager-api/internal/utils"
 )
 
 type BaseInfo struct {
-	ID         string `json:"id"`
-	CreateTime int64  `json:"create_time"`
-	UpdateTime int64  `json:"update_time"`
+	ID         interface{} `json:"id"`
```
Comment (tokers) [inline api/internal/core/entity/entity.go:27]: Why not using `json.Number`?

Comment (nic-chen) [reply api/internal/core/entity/entity.go:27]: Hi, @tokers  
we support id as any string not only number.

Comment (tokers) [reply api/internal/core/entity/entity.go:27]: Oops, i misunderstood it.

**Comment by tokers** [comment]
LGTM

**Review by tokers — APPROVED** [review]

**Pull request merged**